Proxy fetch and XMLHttpRequest through agent frame

jnicklas opened the pull request:
This makes it possible to use mocking libraries like mirage from the test suite. By mocking XMLHttpRequest in the agent, we're also automatically using those mocks in the app/harness.

jnicklas committed:
Proxy fetch and XMLHttpRequest through agent frame

This makes it possible to use mocking libraries like mirage from the test suite. By mocking XMLHttpRequest in the agent, we're also automatically using those mocks in the app/harness.

jnicklas commented:
I'd ideally like to add some tests for this, but I'm not really sure how and where they'd go?

cowboyd commented:
I don't imagine that we'd want to load the harness in anything but an iframe, but even uttering those words makes me feel like it's an inevitability at some point. Since for top-level frames, `window.parent` is actually a reference to itself, this indirection would result in an infinite loop. What if we wrapped these in a guard?

```js
if (window.parent !== window) {
  // implement redirect
}
```

jnicklas commented:
Yeah, good catch. Good to be defensive, and "surely it'll never..." seems to always come back to bite you in the end.

jnicklas committed:
Inline createHarness

It's just confusing having this in a separate file

jnicklas committed:
Test fetch proxying

jnicklas committed:
Only proxy requests if running inside iframe

jnicklas committed:
Don't minify generated manifest

Makes it easier to debug

jnicklas committed:
Quox parcel options

jnicklas committed:
Don't run parcel in agent test suite

Instead use the prepacked package for tests. This makes the tests faster, simpler and less error-prone.
